packages/core-ethereum/src/ethereum.ts

```diff
@@ -154,12 +156,14 @@ export async function createChainWrapper(
       // compute tx hash and save to initiated tx list in tx manager
       const initiatedHash = utils.keccak256(signedTx)
       transactions.addToQueuing(initiatedHash, { nonce, gasPrice }, essentialTxPayload)
-
+      // with let indexer to listen to the tx
+      removeListener = handleTxListener(initiatedHash)
       // 4. send transaction to our ethereum provider
       transaction = await provider.sendTransaction(signedTx)
     } catch (error) {
       log('Transaction with nonce %d failed to sent: %s', nonce, error)
       nonceLock.releaseLock()
+      removeListener()
```

Comment:
If `removeListener` is undefined, this leads to an unhandled exception

Reply:
Close this discussion due to changes in the implementation